base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
```

Comment:
# Optional:
- I would advise you to add `attr_accessor :nameable`  because nameable is both a getter and a setter.
- Here is the 👉  [Link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md) to learn more